api/src/main/java/org/openmrs/module/bedmanagement/dao/impl/BedManagementDaoImpl.java

```diff
@@ -95,11 +96,20 @@ public BedPatientAssignment getBedPatientAssignmentByUuid(String uuid) {
 	@Override
 	public List<BedPatientAssignment> getBedPatientAssignmentByEncounter(String encunterUuid, boolean includeEnded) {
 		Session session = sessionFactory.getCurrentSession();
-		List<BedPatientAssignment> bpaList = (List<BedPatientAssignment>) session
-		        .createQuery("select bpa from BedPatientAssignment bpa " + "inner join bpa.encounter enc "
-		                + "where enc.uuid = :encounterUuid AND " + "(bpa.endDatetime IS NULL OR :includeEnded IS TRUE) AND "
-		                + "(bpa.voided IS FALSE) " + "order by bpa.startDatetime DESC")
-		        .setParameter("encounterUuid", encunterUuid).setParameter("includeEnded", includeEnded).list();
+		List<BedPatientAssignment> bpaList;
+		FlushMode flushMode = session.getHibernateFlushMode();
```

Comment:
I remember doing this exact thing in some places, in the past. Was this problem triggered by a switch to a newer version of OpenMRS core?